pkg/git/gogit.go

```diff
@@ -276,6 +277,34 @@ func (g *GoGit) Head() (string, error) {
 	return head.Hash().String(), nil
 }
 
+func (g *GoGit) ValidateAccess(ctx context.Context, url string, branch string) error {
+
+	path, err := ioutil.TempDir("", "temp-src")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(path)
+
+	branchRef := plumbing.NewBranchReferenceName(branch)
+	_, err = gogit.PlainCloneContext(ctx, path, true, &gogit.CloneOptions{
+		URL:           url,
```

Comment:
@JamWils  you mean validate the format of the input value? like the structure that defines what an actual URL is? Right now the command fails like this `Error: failed to add the app : could not generate deploy key: could not get owner from url notvalidURL.git` with an invalid url value, which might seem ok because it describes(maybe no too clearly) what the problem is.

Reply:
I guess I meant more, do we allow any arbitrary URL to be passed into here or do we have limits.  E.g. do we have only specific hosts we want to let through, etc.  There is the url.URL parse function that could check that it is simply a valid URL in the first place.  Make sense?

Reply:
yes, it does. We currently allow any arbitrary value for the URL. It doesn't exist any validation at this moment.

Reply:
We chatted f2f and @josecordaz is going to validate the URL in another location.